From: Neil Horman <nhorman@redhat.com>
Date: Mon, 3 Nov 2008 10:07:39 -0500
Subject: [crypto] fix sha384 blocksize definition
Message-id: 20081103150739.GC31151@hmsendeavour.rdu.redhat.com
O-Subject: [RHEL5.4 PATCH] crypto: fix sha384 blocksize definition (bz 469167)
Bugzilla: 469167
RH-Acked-by: Herbert Xu <herbert.xu@redhat.com>
RH-Acked-by: Mikulas Patocka <mpatocka@redhat.com>
RH-Acked-by: David Miller <davem@redhat.com>

Hey-
	Backport of Herbert Xu's upstream patch correcting the blocksize
definition for SHA384.  Satisfies bz 469167

Regards
Neil

diff --git a/crypto/sha512.c b/crypto/sha512.c
index 2dfe7f1..15eab9d 100644
--- a/crypto/sha512.c
+++ b/crypto/sha512.c
@@ -24,7 +24,7 @@
 
 #define SHA384_DIGEST_SIZE 48
 #define SHA512_DIGEST_SIZE 64
-#define SHA384_HMAC_BLOCK_SIZE  96
+#define SHA384_HMAC_BLOCK_SIZE 128
 #define SHA512_HMAC_BLOCK_SIZE 128
 
 struct sha512_ctx {
